traversal/procedure/ProcedureVertex.java

```diff
@@ -157,69 +160,74 @@ public boolean overlaps(Thing other, Traversal.Parameters params) {
         }
 
         @Override
-        public Forwardable<? extends ThingVertex, Order.Asc> iterator(GraphManager graphMgr, Traversal.Parameters parameters) {
-            if (props().hasIID()) return iterateAndFilterFromIID(graphMgr, parameters);
-            else return iterateAndFilterFromTypes(graphMgr, parameters);
+        public Forwardable<? extends ThingVertex, Order> iterator(GraphManager graphMgr, Traversal.Parameters parameters,
+                                                                  Modifiers.Sorting sorting) {
+            Order order = sorting.isAscending(id()) ? ASC : DESC;
```

Comment:
convert sorting direction into iterator ordering, which we use throughout this class in order to get an iterator ordered in the right way